store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/util/ApplyViewToElementsFunction.java

```diff
@@ -120,6 +145,55 @@ private static void validate(final Map<String, Object> context) {
         }
     }
 
+    private static void updateEntitiesViewFilterWithSchemaValidation(final Schema schema, final View view, final View.Builder updatedView) {
+        final Map<String, SchemaEntityDefinition> elements = schema.getEntities();
+        for (final Map.Entry<String, ? extends SchemaElementDefinition> schemaElement : elements.entrySet()) {
+
+            final String elementKey = schemaElement.getKey();
+            final ViewElementDefinition updatedViewElementDef = getUpdatedViewElementDef(elementKey, schemaElement.getValue(), view);
+
+            updatedView.entity(elementKey, updatedViewElementDef);
+        }
+    }
+
+    private static void updateEdgesViewFilterWithSchemaValidation(final Schema schema, final View view, final View.Builder updatedView) {
+        final Map<String, SchemaEdgeDefinition> elements = schema.getEdges();
+        for (final Map.Entry<String, ? extends SchemaElementDefinition> schemaElement : elements.entrySet()) {
+
+            final String elementKey = schemaElement.getKey();
+            final ViewElementDefinition updatedViewElementDef = getUpdatedViewElementDef(elementKey, schemaElement.getValue(), view);
+
+            updatedView.edge(elementKey, updatedViewElementDef);
+        }
+    }
```

Comment:
If anyone let me know how I can squash these two into 1 method I'd love that.
`schema.getEdges();` and ` updatedView.edge(elementKey, updatedViewElementDef)`

The only way I can thing is return crappy map.

Reply:
As a thought this might be a good place to use inline lambdas instead you could replace both these methods with just:

```java
schema.getEntities().entrySet().forEach(entity -> {
    if (entity.getValue().hasValidation()) {
        updatedView.entity(entity.getKey(), getUpdatedViewElementDef(entity.getValue(), view, entity.getKey()));
    }
});

schema.getEdges().entrySet().forEach(edge -> {
    if (edge.getValue().hasValidation()) {
        updatedView.edge(edge.getKey(), getUpdatedViewElementDef(edge.getValue(), view, edge.getKey()));
    }
});
```

Reply:
Thank you. I tried lambdas but I couldn't get it play nicely on Friday. but I was trying to pass the a lambda function in to execute within the loop. Completely forgot I could have just stream a foreach and not bothered with the for loop anymore. Silly me. a tired Friday I guess. I just knew it could be tidier. I went with returning a crappy map which you probably have seen. I'll whack this in.

Reply:
I must have been super tired coz I didn't push my map changes on Friday either.